pkg/apis/spinnaker/v1alpha2/spinnakeraccount_types.go

```diff
@@ -27,19 +28,40 @@ type AccountPermissions map[Authorization][]string
 // SpinnakerAccountSpec defines the desired state of SpinnakerAccount
 // +k8s:openapi-gen=true
 type SpinnakerAccountSpec struct {
-	Enabled     bool               `json:"enabled"`
-	Type        AccountType        `json:"type"`
-	Validation  ValidationSetting  `json:"validation"`
+	Enabled bool        `json:"enabled"`
+	Type    AccountType `json:"type"`
+	// +optional
+	Validation ValidationSetting `json:"validation"`
+	// +optional
 	Permissions AccountPermissions `json:"permissions"`
-	Auth        FreeForm           `json:"auth,omitempty"`
-	Env         FreeForm           `json:"env,omitempty"`
-	Settings    FreeForm           `json:"settings,omitempty"`
+	// +optional
+	Kubernetes *KubernetesAuth `json:"kubernetes,omitempty"`
```

Comment:
It seems strange to have provider specific information here that only works for Kubernetes, when there is a generic `Settings` field and a `Type` field. Why not have then all Kubernetes account settings here and not just the auth?, and structs for AWS, GCP, Jenkins and so on...

Reply:
That's to have a more typed spec. That's also how we inline the kubeconfig. Other providers won't have as many options but I'd still like to make them typed instead of having a hodgepodge of settings.

Reply:
Maybe I'm not seeing the full picture here and this is going to change, but I think it's confusing to tell people that all account configuration settings belong to a generic free format section shared between all accounts, while just one setting, of one specific account type, has its own special field handled as a special case. 

We could have all kubernetes settings included in the kubernetes field as well, a free format section for everything and a kubeconfig or auth section for the kubeconfig setting:
```
spec:
  type: kubernetes
  kubernetes:
    auth:
      kubeconfigfile: XXX
    settings: <free format>
```
In this way we can add more types accounts as we need them by having just one level or two of strong typing.

Reply:
Ah I see. To be clear we're talking of:
```
spec:
  type: kubernetes
  permissions: {} 
  kubernetes:
      kubeconfigfile: XXX
    settings: <free format>
```
vs
```
spec:
  type: kubernetes
  permissions: {} 
  kubernetes:
      kubeconfigfile: XXX
  settings: <free format>
```

We have other shared top level fields so I'm not sure why settings is special but maybe I've been looking at it for too long. Let's get some feedback.